src/garbage_collector.rs

```diff
@@ -419,21 +532,17 @@ impl GarbageCollectorOrchestrator {
         );
     }
 
-    fn options_to_ticker(
-        options: Option<&GarbageCollectorDirectoryOptions>,
-    ) -> (crossbeam_channel::Receiver<Instant>, DirGcStatus) {
-        options.map_or(
-            (crossbeam_channel::never(), DirGcStatus::Done),
-            |opts| match opts.execution_mode {
-                GcExecutionMode::Once => {
-                    (crossbeam_channel::at(Instant::now()), DirGcStatus::OneMore)
-                }
-                Periodic(duration) => (
-                    crossbeam_channel::tick(duration),
-                    DirGcStatus::Indefinite(duration),
-                ),
-            },
-        )
+    async fn run_gc_task<T: GcTask>(&self, task: &mut T) {
+        debug!(
+            "Scheduled garbage collection attempt for {}.",
+            task.resource()
+        );
+        if let Err(e) = self.remove_expired_checkpoints().await {
```

Comment:
Nice refactor! One idea might be to let this function return the current manifest, which could then be passed through to `collect`. I think each task has to read the current manifest anyway.

Reply:
I like the idea and I tried to make this change, but it seems like each of these needs something slightly different:

1. expiring removed checkpoints needs just the latest manifest
2. `WalGcTask` and `CompactedGcTask` need all the active manifests
3. `ManifestGcTask` needs all manifests (including inactive, in order to clean them up)

I think it is possible to issue fewer reads to S3 if we do further refactoring, but I think it ends up complicating things a bit and I have to pass a lot of references around.

Let's leave this as a potential follow-up refactor.